components/overview/TableOfContents.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <nav class="table-of-contents">
+    <AppLink
+      v-for="element in elements"
+      :key="element.label"
+      class="table-of-contents__element"
+      :class="{ 'table-of-contents__element_secondary': element.isSecondary }"
+      :url="element.url"
+    >
+      {{ element.label }}
+    </AppLink>
+  </nav>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+import AppLink from '~/components/ui/AppLink.vue'
+
+@Component({
+  components: { AppLink }
+})
+export default class extends Vue {
+  elements = [
```

Comment:
In order to reduce coupling between the presentational and content aspects of the component, let's pass the elements from the consumer. This way, the kind of relationships derived from the `id` of the sections and the `url`s are also in the same place, improving the cohesiveness of the content.